test/rulesets.jl

```diff
@@ -131,3 +131,20 @@ end
     @test isnothing(@rule(f(1)(a) => 2)(sin(a)))
     @test @rule($(f(1))(a) => 2)(sin(a)) == 2
 end
+
+@testset "where" begin
+    expected = Meta.parse("f(~x) ? ~x + ~y : nothing")
+    @test SymbolicUtils.rewrite_rhs(:((~x + ~y) where f(~x))) == expected
+
+    @syms a b
+    f(x) = x === a
+    r = @rule ~x => ~x where f(~x)
+    @eqtest r(a) == a
+    @test isnothing(r(b))
```

Comment:
`isnothing` is slower than `=== nothing`

Reply:
Ok, I'll change this. Please do note that there are currently 5 occurrences of `isnothing` in `src` and 3 in `test/rulesets.jl`. Unlike it's usage in tests, the occurrences in `src/` will affect production performance. Also, due to https://github.com/JuliaLang/julia/pull/38905, the performance difference between `x === nothing` and `isnothing(x)` got even smaller.

Reply:
The linked issue only got into Julia 1.7 as far as I can see, so you probably have a good point given that Julia 1.6 is still very likely to become LTS